lib/screens/calendar/calendar.dart

```diff
@@ -0,0 +1,841 @@
+  const HighLightClose({
+    Key? key,
+  }) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    return Container(
+      margin: EdgeInsets.only(top: 10, bottom: 10, right: 6),
+      decoration: BoxDecoration(
+          color: Color.fromRGBO(110, 176, 230, 0.15),
+          borderRadius: BorderRadius.only(
+            bottomRight: Radius.circular(40),
+            topRight: Radius.circular(40),
+          )),
+    );
+  }
+}
+
+class HighLightMiddle extends StatelessWidget {
+  const HighLightMiddle({
+    Key? key,
+  }) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    return Container(
+      margin: EdgeInsets.only(top: 10, bottom: 10),
+      decoration: BoxDecoration(
+        color: Color.fromRGBO(110, 176, 230, 0.15),
+      ),
+    );
+  }
+}
+
+class HighLightSingle extends StatelessWidget {
+  const HighLightSingle({
+    Key? key,
+  }) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    return Container(
+      margin: EdgeInsets.symmetric(horizontal: 6, vertical: 10),
+      decoration: BoxDecoration(
+          color: Color.fromRGBO(110, 176, 230, 0.15),
+          borderRadius: BorderRadius.circular(40)),
+    );
+  }
+}
+
+class _EntriesSlot extends StatelessWidget {
+  final snapshot;
+
+  _EntriesSlot(this.snapshot);
+
+  String calculateDuration(String st, String et) {
+    Duration duration = DateTime.parse(et).difference(DateTime.parse(st));
+    String hour = duration.inHours % 24 == 0
+        ? ''
+        : '${(duration.inHours % 24).toString()}h';
+    String minutes = duration.inMinutes % 60 == 0
+        ? ''
+        : '${(duration.inMinutes % 60).toString()}m';
+
+    return '${hour} ${minutes}';
+  }
+
+  bool isClassOfEpisodeCalendarEntry(var input) {
+    return (input is Fragment$CalendarDay$entries$$EpisodeCalendarEntry);
+  }
+
+  pushToEpisodePage(BuildContext context, var id) {
+    context.router.push(EpisodeScreenRoute(episodeId: id));
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    var eventsList = snapshot.data?.day.events;
+    var entriesList = snapshot.data?.day.entries;
+    return Column(
+      children: <Widget>[
+        if (entriesList != null && entriesList.isNotEmpty) ...[
+          (eventsList != null)
+              ? Padding(
+                  padding: EdgeInsets.symmetric(horizontal: 16),
+                )
+              : Text('ingen bilder'),
+          for (var i = 0; i < entriesList!.length; i++)
+            GestureDetector(
+              onTapUp: (details) {
+                isClassOfEpisodeCalendarEntry(entriesList[i])
+                    ? pushToEpisodePage(context, entriesList[i].id)
+                    : null;
+              },
+              child: Container(
+                padding: EdgeInsets.symmetric(horizontal: 8, vertical: 16),
+                decoration: BoxDecoration(
+                  color: (DateTime.parse(entriesList[i].end)
```

Comment:
This condition is copy-pasted many times. You can make it cleaner by using a variable